src/common/webrtc/api.rs

```diff
@@ -0,0 +1,463 @@
+        candidate: &Candidate,
+        ufrag: String,
+        uuid: String,
+    ) -> Result<(), WebRTCError> {
+        let answer = AnswerResponse {
+            uuid,
+            stage: Some(answer_response::Stage::Update(AnswerResponseUpdateStage {
+                candidate: Some(IceCandidate {
+                    candidate: candidate.to_string(),
+                    sdp_mid: Some("".to_owned()),
+                    sdpm_line_index: Some(0),
+                    username_fragment: Some(ufrag),
+                }),
+            })),
+        };
+        match self.signaling_tx.send_message(answer) {
+            Err(_) => Err(WebRTCError::SignalingDisconnected()),
+            Ok(_) => Ok(()),
+        }
+    }
+    pub(crate) async fn next_remote_candidate(&mut self) -> Result<Option<Candidate>, WebRTCError> {
+        match self.signaling_rx.next().await {
+            None => Err(WebRTCError::SignalingDisconnected()),
+            Some(req) => {
+                if let Some(stage) = req.stage {
+                    match stage {
+                        answer_request::Stage::Update(c) => {
+                            if let Some(c) = c.candidate {
+                                log::error!("received candidaate {}", c.candidate);
+                                let c = c.candidate.try_into().ok();
+                                return Ok(c);
+                            } else {
+                                log::error!("received no candidates with this update request");
+                                return Ok(None);
+                            }
+                        }
+                        answer_request::Stage::Error(s) => {
+                            if let Some(status) = s.status {
+                                return Err(WebRTCError::SignalingError(status.message));
+                            }
+                            return Err(WebRTCError::SignalingError("unknown".to_owned()));
+                        }
+                        answer_request::Stage::Done(_) => {
+                            return Ok(None);
+                        }
+                        _ => {
+                            return Err(WebRTCError::SignalingError("unexpected stage".to_owned()))
+                        }
+                    }
+                }
+                Ok(None)
+            }
+        }
+    }
+    pub fn send_done(&mut self, uuid: String) -> Result<(), WebRTCError> {
+        let answer = AnswerResponse {
+            uuid,
+            stage: Some(answer_response::Stage::Done(AnswerResponseDoneStage {})),
+        };
+        match self.signaling_tx.send_message(answer) {
+            Err(_) => Err(WebRTCError::SignalingDisconnected()),
+            Ok(_) => Ok(()),
+        }
+    }
+}
+
+#[cfg(feature = "native")]
+type Executor<'a> = NativeExecutor<'a>;
+#[cfg(feature = "esp32")]
+type Executor<'a> = Esp32Executor<'a>;
+
+pub struct WebRTCApi<'a, S, D> {
+    executor: Executor<'a>,
```

Comment:
I like leaving it as a single letter since it quickly  let a reader know it's a generic vs using camelcase in which case it might be a type or collide with a type we want to add later.

Reply:
fine with me if that is convention!